admin/javascript/src/components/form-action/index.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import SilverStripeComponent from 'silverstripe-component';
+
+class FormActionComponent extends SilverStripeComponent {
+  constructor(props) {
+    super(props);
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  render() {
+    const props = {
+      type: this.props.type,
+      className: this.getButtonClasses(),
+      disabled: this.props.disabled,
+    };
+
+    if (typeof this.props.id !== 'undefined') {
+      props.id = this.props.id;
+    }
+
+    return (
+      <button {...props}>
+        {this.getLoadingIcon()}
+        {this.props.label}
+      </button>
+    );
+  }
+
+  /**
+   * Returns the necessary button classes based on the given props
+   *
+   * @returns string
+   */
+  getButtonClasses() {
+    const buttonClasses = ['btn'];
+
+    // Add 'type' class
+    buttonClasses.push(`btn-${this.props.style}`);
+
+    // If there is no text
+    if (typeof this.props.label === 'undefined') {
+      buttonClasses.push('no-text');
+    }
+
+    // Add icon class
+    if (typeof this.props.icon !== 'undefined') {
+      buttonClasses.push(`font-icon-${this.props.icon}`);
+    }
+
+    // Add loading class
+    if (this.props.loading === true) {
+      buttonClasses.push('btn--loading');
```

Comment:
I guess its a "modifier" in BEM speak, so I'd say its deliberate (http://getbem.com/naming/)